infra/lambda/lambda_function.py

```
@@ -167,6 +167,81 @@ def lambda_handler(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
+                        results.append({"job_id": job_id, "status": "blocked" if is_blocked else "completed"})
+
+                    except Exception as e:
+                        # Mark job as failed for visibility; message retries are handled by SQS/Lambda.
+                        try:
+                            store.update_job_status(
+                                job_id=job_id_for_error,
+                                status="failed",
+                                last_error=str(e)[:2000],
+                            )
+                        except Exception:
+                            pass
```

Ruff (0.15.7)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)